web/src/redux/githubID/githubID.reducer.ts

```diff
@@ -18,45 +23,80 @@ const GithubIDBaseState = {
 };
 
 // IMPORTANT:  In this redux state,the first elememt (index 0) is PO, others are TL
-const INITIAL_STATE: GithubIdBaseInterface[] = [GithubIDBaseState, GithubIDBaseState];
+const INITIAL_STATE: GithubIDInitialState = {
+  productOwner: [{ ...GithubIDBaseState }],
+  technicalLeads: [{ ...GithubIDBaseState }, { ...GithubIDBaseState }],
+};
 
-const githubIDReducer = (state = INITIAL_STATE, action: any) => {
-  switch (action.type) {
-    case GithubIDActionTypes.NEW_GITHUB_ID_ENTRY:
-      return [...state, GithubIDBaseState];
+interface GithubUserInputAction {
+  type: string;
+  payload: {
+    persona: 'productOwner' | 'technicalLeads';
+    inputValue: string;
+    position: number;
+  };
+}
+interface GithubIDUserBasicAction {
+  type: string;
+  payload: {
+    persona: 'productOwner' | 'technicalLeads';
+    position: number;
+  };
+}
 
+interface GithubIDUserStore {
+  type: string;
+  payload: {
+    persona: 'productOwner' | 'technicalLeads';
+    position: number;
+    data: object;
+  };
+}
+type Action = GithubUserInputAction & GithubIDUserBasicAction & GithubIDUserStore;
+
+const githubIDReducer = (state = INITIAL_STATE, action: Action) => {
+  switch (action.type) {
     case GithubIDActionTypes.GITHUB_USERS_REQUEST:
-      const newArrayForUserRequest = [...state];
-      newArrayForUserRequest[action.payload] = {
-        ...state[action.payload],
+      const { persona: userRequestPersona, position: userRequestPosition } = action.payload;
+      const newStateForUserRequest = { ...state };
+      newStateForUserRequest[userRequestPersona][userRequestPosition] = {
+        ...newStateForUserRequest[userRequestPersona][userRequestPosition],
         isLoading: true,
         everFetched: false,
       };
-      return newArrayForUserRequest;
+
+      return newStateForUserRequest;
 
     case GithubIDActionTypes.GITHUB_USER_EXISTS:
-      const newArrayForUserExist = [...state];
-      newArrayForUserExist[action.payload] = {
-        ...state[action.payload],
+      const { persona: userExistPersona, position: userExistPosition } = action.payload;
+      const newArrayForUserExist = { ...state };
```

Comment:
```suggestion
      const newState = { ...state };
```

Reply:
be mindful of your variable naming. the object is not even an Array. *slaps wrist*